docs/reference/modules/coordination.asciidoc

```diff
@@ -0,0 +1,497 @@
+  ]
+}
+--------------------------------------------------
+// CONSOLE
+
+This only needs to occur once, on a single master-eligible node in the cluster,
+but for robustness it is safe to repeatedly call `POST /_cluster/bootstrap`,
+and to call it on different nodes concurrently. However **it is vitally
+important** to use the same bootstrap warrant in each call.
+
+WARNING: You must pass the same bootstrap warrant to each call to `POST
+/_cluster/bootstrap` in order to be sure that only a single cluster forms
+during bootstrapping and therefore to avoid the risk of data loss.
+
+The simplest and safest way to construct a bootstrap warrant is to use the `GET
+/_cluster/bootstrap` API:
+
+[source,js]
+--------------------------------------------------
+# Immediately return a bootstrap warrant based on the nodes discovered so far
+GET /_cluster/bootstrap
+# Wait until the node has discovered at least 3 nodes, or 60 seconds has elapsed,
+# and then return the resulting bootstrap warrant
+GET /_cluster/bootstrap?wait_for_nodes=3&timeout=60s
+--------------------------------------------------
+// CONSOLE
+
+This API returns a properly-constructed bootstrap warrant that is ready to pass
+to the `POST /_cluster/bootstrap` API.  It includes all of the master-eligible
+nodes that the handling node has discovered via the gossip-based discovery
+protocol, and returns an error if fewer nodes have been discovered than
+expected.
+
+It is also possible to construct a bootstrap warrant manually and to specify
+the initial set of nodes in terms of their names alone, rather than including
+their IDs too:
+
+[source,js]
+--------------------------------------------------
+# send the bootstrap warrant back to the cluster
+POST /_cluster/bootstrap
+{
+  "master_nodes":[
+    {"name":"master-a"},
+    {"name":"master-b"},
+    {"name":"master-c"}
+  ]
+}
+--------------------------------------------------
+// CONSOLE
+
+It is safer to include the node IDs, in case two nodes are accidentally started
+with the same name.
+
+This process is implemented in the `elasticsearch-bootstrap-cluster`
+command-line tool:
+
+[source,txt]
+--------------------------------------------------
+$ bin/elasticsearch-bootstrap-cluster --node http://10.0.12.1:9200/ \
+    --node http://10.0.13.1:9200/ --node https://10.0.14.1:9200/
+--------------------------------------------------
+
+The arguments to this tool are the addresses of (some, preferably all, of) its
+master-eligible nodes. The tool will construct a bootstrap warrant and then
+bootstrap the cluster, retrying safely if any step fails.
+
+[float]
+=== Unsafe disaster recovery
+
+In a disaster situation a cluster may have lost half or more of its
+master-eligible nodes and therefore be in a state in which it cannot elect a
+master. There is no way to recover from this situation without risking data
+loss, but if there is no other viable path forwards then this may be necessary.
```

Comment:
I wonder if we should be more explicit here, making it clear that it's not only the loss of metadata (cluster state) but possibly the loss of documents. "data" can be misinterpreted here as the cluster state.